tests/metawolf/utils.py

```diff
@@ -0,0 +1,179 @@
+}
+
+MOCK_SESSION_JSON = {
+    'last_active_session': 'session_test',
+    'session_test': {
+        'last_active_recipe': 'recipe_test',
+        'recipe_test': {
+            'session_test-recipe_test-param_name': MOCK_SESSION_SETTABLE_JSON
+        }
+    }
+}
+
+MOCK_DFTIMEWOLF_RECIPE = {
+    'recipe_test': resources.Recipe(
+        description='foo',
+        contents={'name': 'recipe_test'},
+        args=[('argument_1_test', '', None), ('--argument_2_test', '', None)]
+    )
+}
+
+
+class MetawolfUtilsTest(unittest.TestCase):
+  """Test utils.py"""
+  # pylint: disable=unused-argument
+
+  def testIsInt(self) -> None:
+    """That whether or not a string is an int."""
+    self.assertTrue(utils.MetawolfUtils().IsInt('1'))
+    self.assertFalse(utils.MetawolfUtils().IsInt('1.4'))
+    self.assertFalse(utils.MetawolfUtils().IsInt('string'))
+
+  def testIsFloat(self) -> None:
+    """That whether or not a string is a float."""
+    self.assertTrue(utils.MetawolfUtils().IsFloat('1'))
+    self.assertTrue(utils.MetawolfUtils().IsFloat('1.4'))
+    self.assertFalse(utils.MetawolfUtils().IsFloat('string'))
+
+  def testStr2Bool(self) -> None:
+    """That whether or not a string can be represented as a boolean."""
+    for s in ['YeS', 'tRue', 'T', 'y']:
+      self.assertTrue(utils.MetawolfUtils().Str2Bool(s))
+    for s in ['nO', 'fAlsE', 'f', 'N']:
+      self.assertFalse(utils.MetawolfUtils().Str2Bool(s))
+    self.assertIsNone(utils.MetawolfUtils().Str2Bool('foo'))
+
+  def testGetType(self) -> None:
+    """Test that the type of a string is inferred correctly."""
+    self.assertEqual(bool, utils.MetawolfUtils().GetType('y'))
+    self.assertEqual(int, utils.MetawolfUtils().GetType('0'))
+    self.assertEqual(float, utils.MetawolfUtils().GetType('1.'))
+
+  def testCastToType(self) -> None:
+    """Test that a string can be cast to the desired type."""
+    self.assertTrue(utils.MetawolfUtils().CastToType('y', bool))
+    self.assertEqual(0, utils.MetawolfUtils().CastToType('0', int))
+    self.assertEqual(1., utils.MetawolfUtils().CastToType('1.', float))
+
+  def testMarshal(self) -> None:
+    """Test that a session settable object marshals correctly."""
+    self.assertEqual(MOCK_SESSION_SETTABLE_JSON,
+                     utils.MetawolfUtils().Marshal(MOCK_SESSION_SETTABLE))
+
+  def testUnmarshal(self) -> None:
+    """Test that a JSON dict of a session settable unmarshalls correctly."""
+    unmarshalled = utils.MetawolfUtils().Unmarshal(MOCK_SESSION_SETTABLE_JSON)
+    self.assertEqual(
+        MOCK_SESSION_SETTABLE.session_id,
+        unmarshalled.session_id)
+    self.assertEqual(
+        MOCK_SESSION_SETTABLE.description,
+        unmarshalled.description)
+    self.assertEqual(
+        MOCK_SESSION_SETTABLE.GetValue(),
+        unmarshalled.GetValue())
+
+  @mock.patch('json.loads')
+  @mock.patch(
+      'builtins.open', new_callable=mock_open, read_data=str(MOCK_SESSION_JSON))
+  @mock.patch('os.path.exists')
```

Comment:
It would be better to just provide test JSON files instead of mocking everything here.

Reply:
Done